examples/vision/barlow_twins.py

```diff
@@ -0,0 +1,990 @@
+"""
+Title: Barlow Twins
+Author: [Abhiraam Eranti](https://github.com/dewball345)
+Date created: 11/4/21
+Last modified: 11/4/21
+Description: A keras implemenation of Barlow Twins(constrastive SSL with redundancy reduction).
+"""
+"""
+# Barlow Twins
+
+**Author:** Abhiraam Eranti<br>
+**Date created:** 11/6/21<br>
+**Last modified:** 11/6/21<br>
+**Description:** A keras implemenation of Barlow Twins(constrastive SSL with redundancy reduction).
+"""
+
+"""
+Paper: [Barlow Twins: Self-Supervised Learning via Redundancy Reduction](https://arxiv.org/abs/2103.03230)
+
+Original Implementation: [facebookresearch/barlowtwins](https://github.com/facebookresearch/barlowtwins)
+
+Notebook by: Abhiraam Eranti
+
+Self-supervised learning is a relatively novel technique in which a model learns from unlabeled data, and is often used when the data is corrupted or if there is very little of it. A practical use for SSL is to create intermediate embeddings that are learned from the data. These embeddings are based on the dataset itself, with similar images having similar embeddings, and vice versa. They are then attached to the rest of the model, which uses those embeddings as information and effectively learns and makes predictions properly. These embeddings, ideally, should contain as much information and insight about the data as possible, so that the model can make better predictions. However, a common problem that arises is that the model creates embeddings that are redundant. For example, if two images are similar, the model will create embeddings that effectively are just a string of 1’s, or some other value that contains repeating bits of information. This is no better than a one-hot encoding or just having one bit as the model’s representations; it defeats the purpose of the embeddings, as they do not learn as much about the dataset as possible. For other approaches, the solution to the problem was to carefully configure the model such that it tries not to be redundant.
```

Comment:
Please reformat the long lines to shorter ones.
No more than 80 characters per line would be ideal.